dot/peerset/peerset.go

```diff
@@ -332,21 +349,24 @@ func (ps *PeerSet) reportPeer(change ReputationChange, peers ...peer.ID) error {
 
 		setLen := ps.peerState.getSetLength()
 		for i := 0; i < setLen; i++ {
-			if ps.peerState.peerStatus(i, pid) == connectedPeer {
-				// disconnect peer
-				err = ps.peerState.disconnect(i, pid)
-				if err != nil {
-					return err
-				}
+			if ps.peerState.peerStatus(i, pid) != connectedPeer {
+				continue
+			}
 
-				ps.resultMsgCh <- Message{
-					Status: Drop,
-					setID:  uint64(i),
-					PeerID: pid,
-				}
-				if err = ps.allocSlots(i); err != nil {
-					return err
-				}
+			// disconnect peer
+			err = ps.peerState.disconnect(i, pid)
+			if err != nil {
+				return fmt.Errorf("cannot disconnect peer %s at set %d: %w", pid, i, err)
```

Comment:
nit the `pid` and `i` context should be added in the `disconnect` method instead of here